### PR TITLE
FDA-207 Rework API Usage Component.

### PR DESCRIPTION
**Pull request by nagkuchipudi**
FDA-207 Rework API Usage Component.

@beardedfinch, Please review and let me know if there are any changes that need to do.

### CODE REVIEW

File: src/components/ApiUsage.jsx [hunk 1]
```diff
@@ -97,27 +113,11 @@ const ApiUsage = (props:tPROPS) => {
       this.onTrackerChanged = this.onTrackerChanged.bind(this)
     }
 
-    componentDidMount () {
+    componentDidMount = () => {
```
Comment (dkrylovsb) [inline src/components/ApiUsage.jsx:116]: This and other similar function declaration changes aren't really necessary. Were you getting errors from your IDE? Set the JavaScript version as follows:
image

File: package.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
   "dependencies": {
     "babel-eslint": "^10.1.0",
     "babel-polyfill": "^6.26.0",
+    "babel-core": "6.26.3",
+    "babel-loader": "8.2.2",
```
Comment (dkrylovsb) [inline package.json:20]: Curious what prompted adding these? Also, we probably should not be setting the version numbers at fixed values.

File: src/components/ApiUsage.jsx (hunk 1 above)
Comment (nagkuchipudi) [reply src/components/ApiUsage.jsx:116]: sure Denis, I see the errors in the IDE and also this blog says to use the arrow functions. 
https://codeburst.io/how-to-not-react-common-anti-patterns-and-gotchas-in-react-40141fe0dcd

Comment (mercurjl) [reply src/components/ApiUsage.jsx:116]: The biggest attraction to using arrow functions like mentioned in this article is that you don't need to bind them in the constructor any more. Make sure you remember to remove the binding statements in the constructor after converting, I still see some there

Comment (joelmercurio-sb) [reply src/components/ApiUsage.jsx:116]: The biggest draw to converting to arrow functions is that they no longer need to be bound in the constructor. You've converted a few like onHighlightChange(), make sure you remember to remove them from the constructor as well, there are a few still there.

That being said, life-cycle functions like componentDidMount should **not** be converted, those should stay a traditional function declaration: componentDidMount ()

Comment (dkrylovsb) [reply src/components/ApiUsage.jsx:116]: 👍

File: src/components/ApiUsage.jsx [hunk 3]
```diff
@@ -140,8 +140,8 @@ const ApiUsage = (props:tPROPS) => {
           minTime = dataz[0][0]
           maxTime = dataz[dataz.length-1][0]
           maxTime.setDate(maxTime.getDate() + 1)
-          max = Math.max(...dataz.map(v => v[1]))
-          min = Math.min(...dataz.map(v => v[1]))
+          max = Math.max(...dataz.map((v, i) => v[1]))
+          min = Math.min(...dataz.map((v, i) => v[1]))
```
Comment (joelmercurio-sb) [inline src/components/ApiUsage.jsx:144]: You added a the index variable into this constructor but never used it

File: src/components/ApiUsage.jsx (hunk 1 above)
Comment (nagkuchipudi) [reply src/components/ApiUsage.jsx:116]: I'll make these changes, Thanks.

**Comment by joelmercurio-sb** [comment]
I think a part of these reworks should also include purging jQuery if possible. The jQuery in this file is supposed to be adding css to input text elements (line 271-273 and a few below) and does some transforms. In the future if we see things like this we can use the styl files in `src > css > components` to make css changes. Also I looked through this briefly and it doesn't look like it does anything at all, the proper jQuery selector for text inputs is supposed to be  $(":text") I believe.

Another thing to keep an eye out for is state objects that never update. As an example the only place this.state.fontFamily is used is in the jQuery referenced above and is never updated or used updated, just read. Values like this shouldn't be in state and should just be constant variables.

**Comment by beardedfinch** [comment]
There are build errors in the [deployed site](http://open.fda.gov-2021-01-07.s3-website-us-east-1.amazonaws.com/about/statistics/) which will need to be addressed.

**Pull request merged**